code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -388,3 +388,65 @@
 
 /obj/item/reagent_containers/glass/get_belt_overlay()
 	return mutable_appearance('icons/obj/clothing/belt_overlays.dmi', "bottle")
+
+//Mortar & Pestle
+
+/obj/item/pestle
+	name = "pestle"
+	desc = "An ancient, simple tool used in conjunction with a mortar to grind or juice items."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "pestle"
+	force = 4
+
+/obj/item/reagent_containers/glass/mortar
+	name = "mortar"
+	desc = "A specially formed bowl of ancient design. It is possible to crush or juice items placed in it using a pestle; however the process, unlike modern methods, is slow and physically exhausting. Alt click to eject the item."
+	icon_state = "mortar"
+	amount_per_transfer_from_this = 10
+	possible_transfer_amounts = list(5, 10, 15, 20, 25, 30, 50)
+	item_flags = NO_MAT_REDEMPTION
+	reagent_flags = OPENCONTAINER
+	spillable = TRUE
+	var/obj/item/grinded
+
+/obj/item/reagent_containers/glass/mortar/AltClick(mob/user)
+	if(grinded)
+		grinded.forceMove(drop_location())
+		grinded = null
+		to_chat(user, "<span class='notice'>You eject the item inside.</span>")
+
+/obj/item/reagent_containers/glass/mortar/attackby(obj/item/I, mob/living/carbon/human/user)
+	..()
+	if(istype(I,/obj/item/pestle))
+		if(grinded)
+			if(user.getStaminaLoss() > 50)
+				to_chat(user, "<span class='warning'>You are too tired to work!</span>")
+				return
+			to_chat(user, "<span class='notice'>You start grinding...</span>")
+			if((do_after(user, 25, target = src)) && grinded)
+				user.adjustStaminaLoss(20)
+				if(grinded.juice_results) //prioritize juicing
+					grinded.on_juice()
+					reagents.add_reagent_list(grinded.juice_results)
+					to_chat(user, "<span class='notice'>You juice [grinded] into a fine liquid.</span>")
+					QDEL_NULL(grinded)
+					return
+				grinded.on_grind()
+				reagents.add_reagent_list(grinded.grind_results)
+				if(grinded.reagents) //food and pills
+					grinded.reagents.trans_to(src, grinded.reagents.total_volume)
+				to_chat(user, "<span class='notice'>You break [grinded] into powder.</span>")
+				QDEL_NULL(grinded)
+				return
+			return
+		else
+			to_chat(user, "<span class='warning'>There is nothing to grind!</span>")
+			return
+	if(grinded)
+		to_chat(user, "<span class='warning'>There is something inside already!</span>")
+		return
+	if(I.juice_results || I.grind_results)
```

Comment:
should this not also have `I.reagents`?

Reply:
Apparently this is just how it's on tgstation.